packages/algorand-js/src/runtime/runtime.ts

```diff
@@ -0,0 +1,215 @@
+    };
+  }
+
+  assertAccountDefined (a?: SDKAccount): SDKAccount {
+    if (a === undefined) {
+      throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
+    }
+    return a;
+  }
+
+  assertAppDefined (appId: number): SSCParams {
+    const globalDelta = this.interpreter.ctx.state.globalApps.get(appId);
+    if (globalDelta === undefined) {
+      throw new TealError(ERRORS.TEAL.APP_NOT_FOUND);
+    }
+    return globalDelta;
+  }
+
+  getAccount (accountIndex: bigint): SDKAccount {
+    let account: SDKAccount | undefined;
+    if (accountIndex === BIGINT0) {
+      const senderAccount = convertToString(this.interpreter.ctx.tx.snd);
+      account = this.interpreter.ctx.state.accounts.get(senderAccount);
+    } else {
+      accountIndex--;
+      checkIndexBound(Number(accountIndex), this.interpreter.ctx.tx.apat);
+      const pkBuffer = this.interpreter.ctx.tx.apat[Number(accountIndex)];
+      account = this.interpreter.ctx.state.accounts.get(convertToString(pkBuffer));
+    }
+    return this.assertAccountDefined(account);
+  }
+
+  /**
+   * Description: fetches global state value for key present app's global data
+   * returns undefined otherwise
+   * @param appId: current application id
+   * @param key: key to fetch value of from local state
+   */
+  getGlobalState (appId: number, key: Uint8Array): StackElem | undefined {
+    const appDelta = this.assertAppDefined(appId);
+    const globalState = appDelta["global-state"];
+
+    const keyValue = globalState.find(schema => compareArray(schema.key, key));
+    const value = keyValue?.value;
+    if (value) {
+      return value?.bytes || BigInt(value?.uint);
+    }
+    return undefined;
+  }
+
+  /**
+   * Description: add new key-value pair or updating pair with existing key in
+   * app's global data for application id: appId, throw error otherwise
+   * @param appId: current application id
+   * @param key: key to fetch value of from local state
+   * @param value: key to fetch value of from local state
+   */
+  updateGlobalState (appId: number, key: Uint8Array, value: StackElem): SSCStateSchema[] {
+    const appDelta = this.assertAppDefined(appId);
+    const globalState = appDelta["global-state"];
+
+    const data = getKeyValPair(key, value); // key value pair to put
+    const idx = globalState.findIndex(schema => compareArray(schema.key, key));
+    if (idx === -1) {
+      globalState.push(data); // push new pair if key not found
+    } else {
+      globalState[idx].value = data.value; // update value if key found
+    }
+    appDelta["global-state"] = globalState; // save updated state
+
+    assertValidSchema(appDelta["global-state"], appDelta["global-state-schema"]); // verify if updated schema is valid by config
+    return globalState;
+  }
+
+  /**
+   * Description: set accounts for context as {address: SDKAccount}
+   * @param accounts: array of account info's
+   */
+  createStatefulContext (accounts: SDKAccount[]): void {
```

Comment:
Please rename to `initializeAccounts`

Reply:
This assumes that we will use this function only for test setup, not for transaction context preparation.

Reply:
ok